ic-factory/src/api.rs

```diff
@@ -216,6 +328,32 @@ pub trait FactoryCanister: Canister + Sized + PreUpdate {
     fn get_idl() -> ic_canister::Idl {
         ic_canister::generate_idl!()
     }
+
+    fn drop_canister(&self, canister_id: Principal) -> AsyncReturn<Result<(), FactoryError>> {
+        Box::pin(async move {
+            let caller = ic_canister::ic_kit::ic::caller();
+            let state_lock = self.factory_state().borrow_mut().lock()?;
+            let actor = self
+                .factory_state()
+                .borrow_mut()
+                .authorize_owner_int(caller)?
+                .drop_canister(canister_id, &state_lock);
+
+            actor.await?;
```

Comment:
Just await the future, I don't see the point in the actor here.
If there is one please add a comment as to why.